cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxActivity.java

```diff
@@ -91,6 +91,10 @@ public void setEnableVirtualButton(boolean value) {
         this.showVirtualButton = value;
     }
 
+    public void setEnableAudioFocusGain(boolean value) {
+        Cocos2dxAudioFocusManager.gainAudioFocus = value;
```

Comment:
I think we should not define a variable `gainAudioFocus` in Cocos2dxAudioFocusManager which will modify the behavior of `registerAudioFocusListener`.
Instead, Cocos2dxActivity should have a flag for determine whether to invoke `registerAudioFocusListener`.

Besides, I prefer the function name `setAudioFocusGainEnabled`